### PR TITLE
Update msdn links and simplify relative links

### LINKED ISSUE #4893: Broken links to string marshaling examples
Within this topic all of the links to examples are broken.  I needed to see an example of marshaling a string by reference.  Unfortunately the documentation is severely broken.  I tried several links, and it always results in a page no longer available error.

https://github.com/dotnet/docs/blob/master/docs/framework/interop/marshaling-strings.md

### PR DESCRIPTION
**Pull request by mairaw**
Fixes #4893 

I've fixed all the http://msdn.microsoft.com links inside the interop folder (and if the same link was used elsewhere I've fixed that too)

Also related to #4504

### CODE REVIEW

File: docs/framework/interop/default-marshaling-behavior.md [hunk 1]
```diff
@@ -47,10 +42,10 @@ BSTR MethodOne (BSTR b) {
...
+ The runtime always uses the **CoTaskMemFree** method to free memory. If the memory you are working with was not allocated with the **CoTaskMemAlloc** method, you must use an **IntPtr** and free the memory manually using the appropriate method. Similarly, you can avoid automatic memory freeing in situations where memory should never be freed, such as when using the **GetCommandLine** function from Kernel32.dll, which returns a pointer to kernel memory. For details on manually freeing memory, see the [Buffers Sample](https://msdn.microsoft.com/library/733c0dd2-12e5-46e6-8de1-39d5b25df024(v=vs.100)).
```
Comment (rpetrusha) [inline docs/framework/interop/default-marshaling-behavior.md]: This is the wrong URL. The correct one is https://msdn.microsoft.com/library/x3txb6xc(v=vs.100).aspx

File: docs/framework/interop/marshaling-different-types-of-arrays.md [hunk 2]
```diff
@@ -92,7 +88,7 @@ An array is a reference type in managed code that contains one or more elements
     int TestArrayOfStructs2 (MYPERSON* pPersonArray, int size);  
     ```  
   
- [PinvokeLib.dll](http://msdn.microsoft.com/library/5d1438d7-9946-489d-8ede-6c694a08f614) is a custom unmanaged library that contains implementations for the previously listed functions and two structure variables, **MYPOINT** and **MYPERSON**. The structures contain the following elements:  
+ [PinvokeLib.dll](https://msdn.microsoft.com/library/5d1438d7-9946-489d-8ede-6c694a08f614) is a custom unmanaged library that contains implementations for the previously listed functions and two structure variables, **MYPOINT** and **MYPERSON**. The structures contain the following elements:
```
Comment (rpetrusha) [inline docs/framework/interop/marshaling-different-types-of-arrays.md]: This needs to be https://msdn.microsoft.com/library/16014d9f-d6bd-481e-83f0-df11377c550f(v=vs.100)

**Review by rpetrusha — APPROVED** [review]
This looks good, @mairaw. There is still one broken link, one link with an incorrect URL, and two self-referencing links that should be removed. Once you fix those, this is ready to merge.

File: docs/framework/interop/runtime-callable-wrapper.md [hunk 3]
```diff
@@ -38,9 +33,9 @@ Accessing COM objects through the runtime callable wrapper
...
   
 ## Marshaling selected interfaces  
- The primary goal of the [runtime callable wrapper](../../../docs/framework/interop/runtime-callable-wrapper.md) (RCW) is to hide the differences between the managed and unmanaged programming models. To create a seamless transition, the RCW consumes selected COM interfaces without exposing them to the .NET client, as shown in the following illustration.  
+ The primary goal of the [runtime callable wrapper](runtime-callable-wrapper.md) (RCW) is to hide the differences between the managed and unmanaged programming models. To create a seamless transition, the RCW consumes selected COM interfaces without exposing them to the .NET client, as shown in the following illustration.  
   
- ![RCW With Interfaces](../../../docs/framework/interop/media/rcwwithinterfaces.gif "rcwwithinterfaces")  
+ ![RCW With Interfaces](media/rcwwithinterfaces.gif "rcwwithinterfaces")
```
Comment (BillWagner) [inline docs/framework/interop/runtime-callable-wrapper.md]: Another instance of "media"

File: docs/framework/interop/runtime-callable-wrapper.md [hunk 4]
```diff
@@ -28,7 +23,7 @@ The common language runtime exposes COM objects through a proxy called the runti
   
  The runtime creates exactly one RCW for each COM object, regardless of the number of references that exist on that object. The runtime maintains a single RCW per process for each object.  If you create an RCW in one application domain or apartment, and then pass a reference to another application domain or apartment, a proxy to the first object will be used.  As the following illustration shows, any number of managed clients can hold a reference to the COM objects that expose INew and INewer interfaces.  
   
- ![RCW](../../../docs/framework/interop/media/rcw.gif "rcw")  
+ ![RCW](media/rcw.gif "rcw")
```
Comment (BillWagner) [inline docs/framework/interop/runtime-callable-wrapper.md]: Earlier, you used "./media", and here "media".

Nit: we should be consistent.

**Review by BillWagner — APPROVED** [review]
Looks good. I only found a nit or two outside of Ron's comments.

:shipit:

File: docs/framework/interop/com-callable-wrapper.md [hunk 5]
```diff
@@ -44,11 +39,11 @@ Accessing .NET objects through COM callable wrapper
  Unlike the .NET client it wraps, the CCW is reference-counted in traditional COM fashion. When the reference count on the CCW reaches zero, the wrapper releases its reference on the managed object. A managed object with no remaining references is collected during the next garbage-collection cycle.  
   
 ## Simulating COM interfaces  
- The [COM callable wrapper](../../../docs/framework/interop/com-callable-wrapper.md) (CCW) exposes all public, COM-visible interfaces, data types, and return values to COM clients in a manner that is consistent with COM's enforcement of interface-based interaction. For a COM client, invoking methods on a .NET Framework object is identical to invoking methods on a COM object.  
+ The [COM callable wrapper](com-callable-wrapper.md) (CCW) exposes all public, COM-visible interfaces, data types, and return values to COM clients in a manner that is consistent with COM's enforcement of interface-based interaction. For a COM client, invoking methods on a .NET Framework object is identical to invoking methods on a COM object.
```
[the remark this answers is not in the document]
Comment (mairaw) [reply docs/framework/interop/com-callable-wrapper.md]: Good catch @rpetrusha! Also removed the acronym definition since we already exposed that earlier.

**Comment by mairaw** [comment]
Thanks @BillWagner and @rpetrusha. I've addressed all your feedback.

**Pull request merged**